# CMS Medicare Health Equity Analysis: SQL Queries

Loads the 2023 county rows into an in-memory SQLite table called `cms_county` and runs the
three queries in `sql/equity_analysis_queries.sql`.

CMS stores suppressed values (fewer than 11 beneficiaries) as `*`, and the numeric columns
load as text. Every query filters `*` out before casting, because SQLite turns
`CAST('*' AS REAL)` into 0, which would silently pull averages and minimums down.

In [1]:
import pandas as pd
import sqlite3
import re

df = pd.read_csv('../data/raw/2014-2023 Medicare Fee-for-Service Geographic Variation Public Use File.zip', low_memory=False)
county_2023 = df[(df['BENE_GEO_LVL'] == 'County') & (df['YEAR'] == 2023)]

conn = sqlite3.connect(':memory:')
county_2023.to_sql('cms_county', conn, if_exists='replace', index=False)
print(pd.read_sql('SELECT COUNT(*) AS counties FROM cms_county', conn))

   counties
0      3198


### Why the `*` filter matters

The same average computed two ways. Without the filter, `*` counts as 0.

In [2]:
print(pd.read_sql("""
    SELECT
        (SELECT AVG(CAST(BENE_AVG_RISK_SCRE AS REAL)) FROM cms_county) AS avg_risk_wrong,
        (SELECT AVG(CAST(BENE_AVG_RISK_SCRE AS REAL)) FROM cms_county
          WHERE BENE_AVG_RISK_SCRE != '*')                            AS avg_risk_correct,
        (SELECT COUNT(*) FROM cms_county WHERE BENE_AVG_RISK_SCRE = '*') AS suppressed_counties
""", conn))

   avg_risk_wrong  avg_risk_correct  suppressed_counties
0        0.939603          0.955134                   52


In [3]:
# Load the three queries from the .sql file (comments stripped, split on semicolons)
sql_text = open('../sql/equity_analysis_queries.sql', encoding='utf-8').read()
sql_text = re.sub(r'--[^\n]*', '', sql_text)
queries = [q.strip() for q in sql_text.split(';') if q.strip()]
print(len(queries), 'queries loaded')

3 queries loaded


## Query 1: Composite need score, top 20 counties

Need score = min-max normalized dual eligibility rate + min-max normalized risk score
(0 to 2, equal weights). Counties with fewer than 500 FFS beneficiaries are excluded so
small counties with noisy rates do not dominate the ranking.

In [4]:
pd.read_sql(queries[0], conn)

,BENE_GEO_DESC,ffs_beneficiaries,dual_eligibility_rate,avg_risk_score,er_visits_per_1000,std_spending_per_capita,need_score
0,SD-Oglala Lakota,980,0.5204,1.74,577.0,21798.68,1.7337
1,TX-Starr,1857,0.5800,1.45,803.0,17395.38,1.5646
2,CA-Imperial,17828,0.5764,1.32,729.0,14150.14,1.4443
3,SD-Todd,730,0.5068,1.32,441.0,17220.95,1.3421
4,KY-Franklin,5337,0.5033,1.19,1019.0,15294.68,1.2219
5,NY-Kings,142677,0.4342,1.27,546.0,14527.70,1.1913
6,TX-Zapata,628,0.4522,1.24,623.0,13095.32,1.1912
7,TX-Willacy,801,0.3808,1.35,604.0,14296.38,1.1837
8,AK-Kusilvak,513,0.7018,0.78,55.0,7960.16,1.1504
9,FL-Miami-Dade,92559,0.3588,1.33,675.0,16635.80,1.1337


**Note:** AK-Kusilvak and AK-Bethel rank high on need but have very low Medicare FFS ER
use (55 and 68 visits per 1,000). Low use here likely reflects care delivered through tribal
health systems, which is not in this file, rather than low need.

## Query 2: ER visits per ER user, by dual eligibility quartile

Separates "more people use the ER" from "each user visits more often".

In [5]:
pd.read_sql(queries[1], conn)

,dual_quartile,counties,min_dual_rate,max_dual_rate,avg_er_visits_per_1000,avg_share_with_er_visit,avg_visits_per_er_user
0,1,781,0.021,0.106,533.1,0.280,1.89
1,2,781,0.106,0.143,587.2,0.295,1.98
2,3,780,0.143,0.192,621.6,0.304,2.04
3,4,780,0.192,0.702,666.2,0.312,2.12


**Result:** From the lowest to the highest quartile, the share of beneficiaries with any
ER visit rises from 28% to 31%, and visits per ER user rise from 1.89 to 2.12. The extra ER
use comes roughly equally from both.

## Query 3: High ER intensity counties (exploratory)

Counties with above-average ER visits per 1,000 and at least 500 FFS beneficiaries.

In [6]:
pd.read_sql(queries[2], conn)

,BENE_GEO_DESC,ffs_beneficiaries,er_visits_per_1000,dual_eligibility_rate,avg_risk_score,std_spending_per_capita,readmission_rate
0,IL-Hardin,721,1535.0,0.2954,1.02,12561.92,0.1921
1,KY-Wolfe,559,1088.0,0.4097,1.09,12703.82,0.2284
2,KY-Franklin,5337,1019.0,0.5033,1.19,15294.68,0.2193
3,TX-Duval,745,1012.0,0.3544,1.33,19747.69,0.1959
4,MN-Mahnomen,576,998.0,0.2934,1.02,13036.31,0.1908
5,TX-Refugio,759,986.0,0.1357,1.17,18603.25,0.1835
6,TX-Reeves,1123,985.0,0.2663,0.88,13524.27,0.1863
7,KY-Cumberland,871,964.0,0.3169,0.88,13040.73,0.1220
8,NC-Bertie,2515,951.0,0.2258,1.03,12114.06,0.2256
9,LA-Washington,4070,946.0,0.2853,1.23,17953.84,0.1829
